### PR TITLE
OCM-7141 | ci: Automate ids 72524/72525/72517

### PR DESCRIPTION
**Pull request by radtriste**
**What this PR does / why we need it**:
Day2 cases automation

**Which issue(s) this PR fixes** 
https://issues.redhat.com/browse/OCM-7141

**Logs**
<details>
<summary>72524/72525</summary>
Will run 1 of 74 specs
time="2024-04-23T13:34:58+02:00" level=info msg="Loaded cluster profile configuration from profile rosa-hcp-ad : map[admin_enabled:false autoscale:true byok:true byovpc:true channel_group:candidate cloud_provider:aws cluster_type:rosa-hcp compute_machine_type:m5.2xlarge compute_replicas:6 etcd_encryption:true fips:false kms_key_arn:true labeling:false major_version:4.15 multi_az:true oidc_config:managed private:false product_id:rosa proxy:true region:us-west-2 sts:true tagging:true unified_acc_role_path:/advanced/ version: version_pattern:latest zones:a,b,c]"
time="2024-04-23T13:34:58+02:00" level=info msg="Running terraform init against the dir /home/tradisso/projects/openshift/terraform-provider-rhcs/tests/tf-manifests/rhcs/clusters/rosa-hcp"
time="2024-04-23T13:35:01+02:00" level=info msg="Running terraform output against the dir: /home/tradisso/projects/openshift/terraform-provider-rhcs/tests/tf-manifests/rhcs/clusters/rosa-hcp"
SSStime="2024-04-23T13:35:04+02:00" level=info msg="Running terraform init against the dir /home/tradisso/projects/openshift/terraform-provider-rhcs/tests/tf-manifests/rhcs/cluster-autoscaler/hcp"
time="2024-04-23T13:35:04+02:00" level=info msg="Running terraform apply against the dir: /home/tradisso/projects/openshift/terraform-provider-rhcs/tests/tf-manifests/rhcs/cluster-autoscaler/hcp with args [-var pod_priority_threshold=-10 -var max_node_provision_time=1h -var resource_limits={\"max_nodes_total\":10} -var cluster_id=2aqhlltt814o50pnqnm3n1hsq1c2s4l1 -var max_pod_grace_period=10]"
time="2024-04-23T13:35:06+02:00" level=info msg="Running terraform output against the dir: /home/tradisso/projects/openshift/terraform-provider-rhcs/tests/tf-manifests/rhcs/cluster-autoscaler/hcp"
time="2024-04-23T13:35:07+02:00" level=info msg="Running terraform output against the dir: /home/tradisso/projects/openshift/terraform-provider-rhcs/tests/tf-manifests/rhcs/cluster-autoscaler/hcp"
time="2024-04-23T13:35:07+02:00" level=info msg="Running terraform apply against the dir: /home/tradisso/projects/openshift/terraform-provider-rhcs/tests/tf-manifests/rhcs/cluster-autoscaler/hcp with args [-var pod_priority_threshold=3 -var max_node_provision_time=60m -var resource_limits={\"max_nodes_total\":20} -var cluster_id=2aqhlltt814o50pnqnm3n1hsq1c2s4l1 -var max_pod_grace_period=5]"
time="2024-04-23T13:35:10+02:00" level=info msg="Running terraform output against the dir: /home/tradisso/projects/openshift/terraform-provider-rhcs/tests/tf-manifests/rhcs/cluster-autoscaler/hcp"
time="2024-04-23T13:35:10+02:00" level=info msg="Running terraform output against the dir: /home/tradisso/projects/openshift/terraform-provider-rhcs/tests/tf-manifests/rhcs/cluster-autoscaler/hcp"
time="2024-04-23T13:35:10+02:00" level=info msg="Running terraform destroy against the dir: /home/tradisso/projects/openshift/terraform-provider-rhcs/tests/tf-manifests/rhcs/cluster-autoscaler/hcp"
args:  /usr/bin/terraform destroy -auto-approve -no-color -var cluster_id=2aqhlltt814o50pnqnm3n1hsq1c2s4l1 -var max_pod_grace_period=5 -var pod_priority_threshold=3 -var max_node_provision_time=60m -var resource_limits={"max_nodes_total":20}
•SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS

Ran 1 of 74 Specs in 15.409 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 73 Skipped
PASS

</details>

<details>
<summary>72517</summary>
Will run 1 of 74 specs
time="2024-04-23T13:32:35+02:00" level=info msg="Loaded cluster profile configuration from profile rosa-hcp-ad : map[admin_enabled:false autoscale:true byok:true byovpc:true channel_group:candidate cloud_provider:aws cluster_type:rosa-hcp compute_machine_type:m5.2xlarge compute_replicas:6 etcd_encryption:true fips:false kms_key_arn:true labeling:false major_version:4.15 multi_az:true oidc_config:managed private:false product_id:rosa proxy:true region:us-west-2 sts:true tagging:true unified_acc_role_path:/advanced/ version: version_pattern:latest zones:a,b,c]"
time="2024-04-23T13:32:35+02:00" level=info msg="Running terraform init against the dir /home/tradisso/projects/openshift/terraform-provider-rhcs/tests/tf-manifests/rhcs/clusters/rosa-hcp"
time="2024-04-23T13:32:38+02:00" level=info msg="Running terraform output against the dir: /home/tradisso/projects/openshift/terraform-provider-rhcs/tests/tf-manifests/rhcs/clusters/rosa-hcp"
SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSStime="2024-04-23T13:32:39+02:00" level=info msg="Loaded cluster profile configuration from profile rosa-hcp-ad : map[admin_enabled:false autoscale:true byok:true byovpc:true channel_group:candidate cloud_provider:aws cluster_type:rosa-hcp compute_machine_type:m5.2xlarge compute_replicas:6 etcd_encryption:true fips:false kms_key_arn:true labeling:false major_version:4.15 multi_az:true oidc_config:managed private:false product_id:rosa proxy:true region:us-west-2 sts:true tagging:true unified_acc_role_path:/advanced/ version: version_pattern:latest zones:a,b,c]"
time="2024-04-23T13:32:40+02:00" level=info msg="Running terraform init against the dir /home/tradisso/projects/openshift/terraform-provider-rhcs/tests/tf-manifests/rhcs/ingresses/hcp"
time="2024-04-23T13:32:41+02:00" level=info msg="Running terraform apply against the dir: /home/tradisso/projects/openshift/terraform-provider-rhcs/tests/tf-manifests/rhcs/ingresses/hcp with args [-var cluster=2aqhlltt814o50pnqnm3n1hsq1c2s4l1 -var listening_method=internal]"
time="2024-04-23T13:32:45+02:00" level=info msg="Running terraform apply against the dir: /home/tradisso/projects/openshift/terraform-provider-rhcs/tests/tf-manifests/rhcs/ingresses/hcp with args [-var cluster=2aqhlltt814o50pnqnm3n1hsq1c2s4l1 -var listening_method=external]"
time="2024-04-23T13:32:49+02:00" level=info msg="Running terraform destroy against the dir: /home/tradisso/projects/openshift/terraform-provider-rhcs/tests/tf-manifests/rhcs/ingresses/hcp"
args:  /usr/bin/terraform destroy -auto-approve -no-color -var cluster=2aqhlltt814o50pnqnm3n1hsq1c2s4l1 -var listening_method=external
time="2024-04-23T13:32:51+02:00" level=info msg="Running terraform apply against the dir: /home/tradisso/projects/openshift/terraform-provider-rhcs/tests/tf-manifests/rhcs/ingresses/hcp with args [-var cluster=2aqhlltt814o50pnqnm3n1hsq1c2s4l1 -var listening_method=external]"
•SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS

Ran 1 of 74 Specs in 18.893 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 73 Skipped
PASS
</details>

**Change type**
- [ ] New feature
- [ ] Bug fix
- [ ] Build
- [x] CI
- [ ] Documentation
- [ ] Performance
- [ ] Refactor
- [ ] Style
- [ ] Unit tests
- [ ] Subsystem tests

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.

### CODE REVIEW

File: tests/e2e/classic_ingress_test.go [hunk 1]
```diff
@@ -40,22 +49,8 @@ var _ = Describe("Ingress updating", ci.NonHCPCluster, func() {
 				Cluster:          clusterID,
 			}
 			err = ingressService.Apply(&args)
-			if profile.GetClusterType().HCP {
```
Comment (xueli181114) [inline tests/e2e/classic_ingress_test.go:43]: If this part is removed, we cannot detect the error message of HCP ingress updating anymore.

Comment (radtriste) [reply tests/e2e/classic_ingress_test.go:43]: Classic and HCP ingresses are now tested separately. Do we want to test incompatibility between classic and hcp specific resources ?

File: tests/e2e/classic_ingress_test.go [hunk 2]
```diff
@@ -91,30 +87,15 @@ var _ = Describe("Ingress updating", ci.NonHCPCluster, func() {
...
-					RouteSelectors:                ingress.RouteSelectors(),
-					RouteNamespaceOwnershipPolicy: string(ingress.RouteNamespaceOwnershipPolicy()),
-					RouteWildcardPolicy:           string(ingress.RouteWildcardPolicy()),
-				}
-				ingressService.Apply(&args)
-			}()
+
 			By("use ocm API to check if ingress config updated")
-			ingress, err = cms.RetrieveClusterIngress(ci.RHCSConnection, clusterID)
+			ingressBefore, err = cms.RetrieveClusterIngress(ci.RHCSConnection, clusterID)
```
[the remark this answers is not in the document]
Comment (radtriste) [reply tests/e2e/classic_ingress_test.go:93]: That was already the case before ... I will correct that.

File: tests/e2e/classic_ingress_test.go (hunk 1 above)
Comment (xueli181114) [reply tests/e2e/classic_ingress_test.go:43]: For the resource, the checkpoint on HCP is not for the resource itself. But to check if errors happen, like HCP cluster reject the ingress updating if terraform provider can exit with backend error. There was a bug before that terraform provider cannot exit with some error happened, it stuck

Comment (radtriste) [reply tests/e2e/classic_ingress_test.go:43]: not sure I got the point.
Should I leave it here ?
I mean, yeah, HCP and Classic have different ingress fields for same API endpoint, but do we want to test cross-fields ?
Then should I add a check on HCP ingress resource testing to make sure that if classic cluster, then it fails ?

Comment (xueli181114) [reply tests/e2e/classic_ingress_test.go:43]: @radtriste Yeah, let's leave it there. It is used to verify the happened error on HCP.

File: tests/e2e/classic_ingress_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (xueli181114) [reply tests/e2e/classic_ingress_test.go:93]: The case before used `defer`, not `AfterEach`. When defer is inited, it picked the original value.

File: tests/e2e/classic_ingress_test.go (hunk 1 above)
Comment (radtriste) [reply tests/e2e/classic_ingress_test.go:43]: Ok. With https://github.com/terraform-redhat/terraform-provider-rhcs/pull/632 I deactivated this case for HCP ingress, I guess I need to set it back also.

File: tests/e2e/classic_ingress_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (radtriste) [reply tests/e2e/classic_ingress_test.go:93]: I see. Will update that

[the remark this answers is not in the document]
Comment (radtriste) [reply tests/e2e/classic_ingress_test.go]: updated

File: tests/e2e/classic_ingress_test.go (hunk 1 above)
Comment (radtriste) [reply tests/e2e/classic_ingress_test.go:43]: restored

**Pull request merged**